libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -0,0 +1,88 @@
+@file:JvmName("NavigationRouteEx")
+package com.mapbox.navigation.base.route
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.DirectionsWaypoint
+import com.mapbox.api.directions.v5.models.RouteOptions
+
+class NavigationRoute(
+    val directionsResponse: DirectionsResponse,
+    val routeIndex: Int,
+    val routeOptions: RouteOptions
+) {
+    val directionsRoute = directionsResponse.routes()[routeIndex]!!
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is NavigationRoute) return false
+
+        if (directionsResponse != other.directionsResponse) return false
+        if (routeIndex != other.routeIndex) return false
+        if (routeOptions != other.routeOptions) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = directionsResponse.hashCode()
+        result = 31 * result + routeIndex
+        result = 31 * result + routeOptions.hashCode()
+        return result
+    }
+
+    override fun toString(): String {
+        return "NavigationRoute(" +
+            "directionsResponse=$directionsResponse, " +
+            "routeIndex=$routeIndex, " +
+            "routeOptions=$routeOptions" +
+            ")"
+    }
+}
+
+fun List<NavigationRoute>.toDirectionsRoutes() = map { it.toDirectionsRoute() }
+
+fun NavigationRoute.toDirectionsRoute(): DirectionsRoute {
+    return directionsResponse.routes()[routeIndex].toBuilder()
+        .requestUuid(directionsResponse.uuid())
+        .routeIndex(routeIndex.toString())
+        .routeOptions(routeOptions)
+        .build()
+}
+
+fun List<DirectionsRoute>.toNavigationRoutes() = map { it.toNavigationRoute() }
+
+@JvmOverloads
+fun DirectionsRoute.toNavigationRoute(
+    routeIndex: Int = routeIndex()?.toInt() ?: 0
```

Comment:
route index must be provided, otherwise it hard to catch problem later
```suggestion
    routeIndex: Int = routeIndex()?.toInt() ?: throw IllegalStateException("route index mustn't be null")
```

Reply:
also, why do we allow providing `routeIndex` outside? It has to be consumed from directionsRoute, doesn't it?

Reply:
We don't really enforce that on the `mapbox-java` level, it is automated only if someone deserializes the whole `DirectionsResponse` but not if they deserialize only `DirectionsRoute`.

Opting to fully enforce the index to be in the object though, I think it shouldn't cause too many problems and makes the SDK more robust.